internal/examples/go.mod

```diff
@@ -0,0 +1,50 @@
+module github.com/open-telemetry/opamp-go/internal/examples
+
+go 1.17
+
+require (
+	github.com/knadh/koanf v1.3.3
+	github.com/oklog/ulid/v2 v2.0.2
+	github.com/open-telemetry/opamp-go v0.1.0
```

Comment:
This is what I think I am supposed to do when splitting an existing directory into a module (reference a newer than current version).